src/System.Private.DataContractSerialization/src/System/Xml/XmlBaseWriter.cs

```diff
@@ -1376,6 +1618,14 @@ private void FlushBase64()
             }
         }
 
+        private async Task FlushBase64Async()
+        {
+            if (_trailByteCount > 0)
+            {
+                await FlushTrailBytesAsync();
+            }
+        }
```

Comment:
This would be more efficient as:
```C#
private Task FlushBase64Async()
{
    return _trailByteCount > 0 ? 
        FlushTrailBytesAsync() :
        Task.CompletedTask;
}
```